### PR TITLE
Adds the Heimdallr Armor Module, a flare launcher

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Adds the Heimdallr Armor Module, a flare launcher

Codewise a gun beacuse apparently its the best way to do it

Fires flares like the experimental armor module

## Why It's Good For The Game

It's cool

## Changelog

:cl:
add: Adds the Heimdallr Armor Module, a flare launcher
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
image
They are PROC_REF now

**Comment by Losenis** [comment]
How does this work for storage? Does it give the armor storage to contain the flares it'd launch?

**Comment by Xander3359** [comment]
> a way to mass-reload by just using a flare box on it.

It's already possible to add this to any storage simply by using var/refill_types 
image

**Comment by Goblinu** [comment]
The flare powercreep continues

**Review by Kuro020 — APPROVED** [review]

**Comment by Naaanii** [comment]
> The flare powercreep continues

Annoy Zaro #12109

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Debatable if should be free but the balance screamers will figure it out

**Comment by 567Turtle** [comment]
FUCK...

**Comment by 567Turtle** [comment]
Rename to Sol, the god of the sun in Norse mythology so I don't have to rename my pr ty :) #12507 [12507](https://github.com/tgstation/TerraGov-Marine-Corps/pull/12507)

**Comment by 567Turtle** [comment]
Nevermind I renamed mine to Freyr

File: code/modules/projectiles/guns/grenade_launchers.dm [hunk 1]
```diff
@@ -122,6 +122,43 @@ The Grenade Launchers
...
+	icon = 'icons/mob/modular/modular_armor_modules.dmi'
+	icon_state = "heimdallr"
+	item_state = "heimdallr"
+	base_gun_icon = "heimdallr"
+	placed_overlay_iconstate = "heimdallr"
+	fire_sound = 'sound/weapons/guns/fire/flare.ogg'
+	fire_rattle = 'sound/weapons/guns/fire/flare.ogg'
+	fire_animation = null
+	flags_equip_slot = null
+	flags_gun_features = GUN_AMMO_COUNTER|GUN_IFF
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/grenade_launchers.dm:137]: why's it IFF?

File: code/modules/projectiles/guns/grenade_launchers.dm [hunk 2]
```diff
@@ -122,6 +122,43 @@ The Grenade Launchers
...
+	flags_item = null
+	max_shells = 7 //codex
+	wield_delay = 0 SECONDS //null
+	aim_slowdown = 0
+	attachable_allowed = list(
+		/obj/item/attachable/shoulder_mount/flare_cannon,
+	)
+	starting_attachment_types = list(/obj/item/attachable/shoulder_mount/flare_cannon)
+	fire_delay = 1 SECONDS
+	max_chamber_items = 7
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/grenade_launchers.dm:147]: max_shells and max_chamber_items say 7, desc says 14.

File: code/modules/projectiles/guns/grenade_launchers.dm [hunk 3]
```diff
@@ -122,6 +122,43 @@ The Grenade Launchers
...
+	default_ammo_type = /obj/item/explosive/grenade/flare
+	allowed_ammo_types = list(
+		/obj/item/explosive/grenade/flare,
+		/obj/item/explosive/grenade/flare/cas,
+	)
+
+/obj/item/weapon/gun/grenade_launcher/multinade_launcher/able_to_fire(mob/user)
```
Comment (Lumipharon) [inline code/modules/projectiles/guns/grenade_launchers.dm:154]: is uh, this not supposed to be at the flare_cannon level?

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**